heroic-core/src/main/java/com/spotify/heroic/http/query/QueryResource.java

```diff
@@ -54,62 +63,84 @@
     private final JavaxRestFramework httpAsync;
     private final QueryManager query;
     private final AsyncFramework async;
+    private final QueryLogger queryLogger;
 
     @Inject
-    public QueryResource(JavaxRestFramework httpAsync, QueryManager query, AsyncFramework async) {
+    public QueryResource(
+        final JavaxRestFramework httpAsync, final QueryManager query, final AsyncFramework async,
+        final QueryLoggerFactory queryLoggerFactory
+    ) {
         this.httpAsync = httpAsync;
         this.query = query;
         this.async = async;
+        this.queryLogger = queryLoggerFactory.create("QueryResource");
     }
 
     @POST
     @Path("metrics")
     @Consumes(MediaType.TEXT_PLAIN)
     public void metricsText(
-        @Suspended final AsyncResponse response, @QueryParam("group") String group, String query
+        @Suspended final AsyncResponse response, @QueryParam("group") String group,
+        @Context final HttpServletRequest servletReq, final String query
     ) {
+        final HttpContext httpContext = CoreHttpContextFactory.create(servletReq);
+        final QueryContext queryContext = QueryContextFactory.create(Optional.empty());
```

Comment:
If we move the factory to QueryContext then this can just be replaced by QueryContext.empty();